### ISSUE TITLE
Does not actually auto-reload pug templates

### ISSUE THREAD

**Issue by peekxc** [11ty/eleventy-dev-server#66]
I have a simple site, something like...

```
_includes/
    base.pug
content/
    index.md
eleventy.config.js
...
```

In my eleventy config, I'm using the dev server and watching via: 

```
eleventyConfig.setServerOptions({
  domDiff: false,
  port: 8080,
  watch: ["_includes/*.pug", "_includes/**/*.pug", "content/**/*.md"]
})
```

When I change the markdown content, the page changes its content and eleventy reports it, as expected: 

```{bash}
[11ty] Copied x files / Wrote 2 files in 0.02 seconds (v2.0.1)
[11ty] Watching…
[11ty] File changed: _includes/base.pug
[11ty] File changed: _includes/base.pug (skips build)
[11ty] Writing docs/index.html from ./content/index.md (liquid)
```

Whenever I change any of my pug templates, however, though the server re-writes the content the changes to the pug templates don't get propagated. For example, just changing commenting out huge include, like e.g. 

```
html(lang="en-US")
	include partials/head.pug	
	include navbar.pug
```

to 

```
html(lang="en-US")
	include partials/head.pug	
	//- include navbar.pug
```

I see no changes on the browser side. This also didn't seem to work going back to the browser sync methodology. I can only assume this is a bug?

**Comment by peekxc** [comment]
I realized that the issue was related to pug partials/includes. Namely, as long as I reload the file that performs the 'include', I get the expected result. 

Sorry for the false positive.

**Comment by vgpena** [comment]
I encountered the same behavior with WebC components. I'd love to have the dev server propagate changes/re-loads upwards when an include is changed. I'm not sure what infrastructure 11ty currently has to determine this, but I'll take a look 👀 Has this already been tried and/or is there some reason it can't be done at the moment?

(Personally, my own 11ty project is small, so I'd be OK with a naïve "rebuild all layouts whenever any include changes" opt-in flag.)

Thanks for filing this issue BTW, I had noticed the pattern but didn't understand _why_ it was happening.
